src/logistration/RegistrationPage.jsx

```diff
@@ -127,21 +236,97 @@ class RegistrationPage extends React.Component {
       nameValid,
       usernameValid,
       passwordValid,
-      countryValid,
     } = this.state;
+
+    const validityMap = REGISTRATION_VALIDITY_MAP;
+    const validStates = [];
+    Object.keys(validityMap).forEach((key) => {
+      const value = validityMap[key];
+      if (value) {
+        const state = camelCase(key);
+        const stateValid = `${state}Valid`;
+        validStates.push(stateValid);
+      }
+    });
+    let extraFieldsValid = true;
+    validStates.forEach((value) => {
+      extraFieldsValid = extraFieldsValid && this.state[value];
+    });
+
     this.setState({
-      formValid: emailValid && nameValid && usernameValid && passwordValid && countryValid,
+      formValid: emailValid && nameValid && usernameValid && passwordValid && extraFieldsValid,
     });
   }
 
-  renderCountryList() {
-    const locale = getLocale();
-    let items = [{ value: '', label: 'Country or Region of Residence (required)' }];
-    items = items.concat(getCountryList(locale).map(({ code, name }) => ({ value: code, label: name })));
-    return items;
+  addExtraFields() {
+    const fields = this.props.formData.fields.map((value) => {
+      const field = value;
```

Comment:
I believe no need to define an extra variable here, you could simply do this:
```suggestion
    const fields = this.props.formData.fields.map((field) => {
```

Reply:
done